src/engine.c

```diff
@@ -869,6 +869,79 @@ ENGINE_ellipse(ENGINE* engine, int64_t x0, int64_t y0, int64_t x1, int64_t y1, u
   ENGINE_blitBuffer(engine, x0, y0);
 }
 
+internal void
+ENGINE_triangle(ENGINE* engine, float x0, float y0, float x1, float y1, float x2, float y2, uint32_t c) {
+  ENGINE_line(engine, x0, y0, x1, y1, c, 1);
+  ENGINE_line(engine, x1, y1, x2, y2, c, 1);
+  ENGINE_line(engine, x2, y2, x0, y0, c, 1);
+}
+
+internal void
+ENGINE_trianglefillFlatBottom(ENGINE* engine, float x0, float y0, float x1, float y1, float x2, float y2, uint32_t c) {
+  float invslope0 = (x1 - x0) / (y1 - y0);
+  float invslope1 = (x2 - x0) / (y2 - y0);
+  float curx0 = x0;
+  float curx1 = x0;
+
+  for (int scanlineY = y0; scanlineY <= y1; scanlineY++) {
+    ENGINE_line(engine, curx0, scanlineY, curx1, scanlineY, c, 1);
+    curx0 += invslope0;
+    curx1 += invslope1;
+  }
+}
+
+internal void
+ENGINE_trianglefillFlatTop(ENGINE* engine, float x0, float y0, float x1, float y1, float x2, float y2, uint32_t c) {
```

Comment:
ENGINE_trianglefillFlatBottom and ENGINE_trianglefillFlatTop look so similar, is there no way to combine them into a single method? (I don't fully grasp the triangle algorithm at the moment)